scripts/views/mapboxView.js

```diff
@@ -373,17 +436,17 @@
   mapboxView.toggleBorders = function () {
     $('.border-toggle').removeClass('active');
     $(this).addClass('active');
-
+    mapboxView.removeSelections();
+    mapboxView.removeListeners();
     if (this.id === 'show-federal-borders') {
-      map.setLayoutProperty('district_border', 'visibility', 'visible');
-
       map.setLayoutProperty('states-house-districts', 'visibility', 'none');
       map.setLayoutProperty('states-senate-districts', 'visibility', 'none');
+      map.setLayoutProperty('district_border', 'visibility', 'visible');
+      mapboxView.addDistrictListener();
     } else if (this.id === 'show-state-borders') {
-      map.setLayoutProperty('district_border', 'visibility', 'none');
-
       map.setLayoutProperty('states-house-districts', 'visibility', 'visible');
```

Comment:
this still needs to have `map.setLayoutProperty('district_border', 'visibility', 'none');` right?

Reply:
hmm, weird. It was working fine without it, but yeah i'll add this line back in.